pkg/analysis_server/test/services/correction/fix_test.dart

```diff
@@ -5943,6 +5943,44 @@ main() {
 ''');
   }
 
+  test_removeEmptyStatement_insideBlock() async {
+    String src = '''
+void foo() {
+  while(true) {
+    /*LINT*/;
+  }
+}
+''';
+    await findLint(src, LintNames.empty_statements);
+
+    await applyFix(DartFixKind.REMOVE_EMPTY_STATEMENT);
+
+    verifyResult('''
+void foo() {
+  while(true) {
+  }
+}
+''');
+  }
+
+  test_removeEmptyStatement_outOfBlock() async {
```

Comment:
You are totally right, I noticed the same and the lint marks thoses cases, I agree with replacing it with `{}` in those cases, I think is the best solution.